core/src/main/java/org/opensearch/sql/ast/expression/QualifiedName.java

```diff
@@ -13,26 +13,55 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Optional;
 import java.util.stream.StreamSupport;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
 import org.opensearch.sql.ast.AbstractNodeVisitor;
+import org.opensearch.sql.data.type.ExprCoreType;
 
 @Getter
 @EqualsAndHashCode(callSuper = false)
 public class QualifiedName extends UnresolvedExpression {
   private final List<String> parts;
 
+  private final Boolean isMetadataField;
+
   public QualifiedName(String name) {
+    this(name, Boolean.FALSE);
+  }
+
+  public static final String METADATA_FIELD_ID = "_id";
+  public static final String METADATA_FIELD_INDEX = "_index";
+  public static final String METADATA_FIELD_SCORE = "_score";
+  public static final String METADATA_FIELD_MAXSCORE = "_maxscore";
+  public static final String METADATA_FIELD_SORT = "_sort";
+  public static final java.util.Map<String, ExprCoreType> METADATAFIELD_TYPE_MAP = new HashMap<>() {
```

Comment:
We may not want to hardcoding this in `core`. Not sure if adding these fields in `OpenSearchIndex.getFieldMapping()` can work for you, however we need to consider meta column in other storage engine. For example, S3 may have `$path`, `$partition` in each row.

Reply:
Sorry, I didn't mean you need to depend on `opensearch` module. OpenSearchIndex is subclass of our `Table` interface. As I understand, `Analyzer` will fetch field list from `Table` during query analysis. I saw you've added `isMetaField` flag, so I don't see why we want to hardcode metadata field list in `core` module.

Reply:
Oh.  Interesting.  I'll see if I can hook that up.